ddsp/training/ddsp_run.py

```diff
@@ -112,6 +119,58 @@ def delay_start():
     logging.info('Waiting for %i second(s)', delay_time)
     time.sleep(delay_time)
 
+def download_from_gstorage(gstorage_path, local_path):
+  """Downloads the bucket.
+
+  Args:
+    gstorage_path: Path to the object inside the bucket that needs to be
+      downloaded. Format: gs://bucket-name/path/to/file.txt
+    local_path: Local path where downloaded file should be stored.
+  """
+  gstorage_path = gstorage_path.strip('gs:/')
+  bucket_name = gstorage_path.split('/')[0]
+  blob_name = os.path.relpath(gstorage_path, bucket_name)
+
+  storage_client = storage.Client()
+
+  bucket = storage_client.bucket(bucket_name)
+  blob = bucket.blob(blob_name)
+
+  blob.download_to_filename(local_path)
+  logging.info(
+      'Downloaded the bucket object %s. Current location: %s',
+      gstorage_path, local_path)
+
+def handle_gstorage_paths(paths):
+  """Handles gstorage paths.
+
+  If gstorage path is provided calls a function which downloads the file.
+  If local path is provided nothing happens.
+
+  Args:
+    paths: Single path or a list of paths.
+
+  Returns:
+    Single local path or a list of local paths.
+  """
+  if isinstance(paths, str):
+    if re.match('gs://*', paths):
+      local_name = os.path.basename(paths)
+      download_from_gstorage(paths, os.path.join(GIN_PATH, local_name))
```

Comment:
GIN_PATH should be passed in as an arg, not a global.

Reply:
Done.